### PR TITLE
[DOC] expanding content in testing section of "adding estimator" developer docs

### PR DESCRIPTION
**Pull request by aiwalter**
#### What does this implement/fix? Explain your changes.
Added pyhton examples to doc of how to run particular tests/estimators

### CODE REVIEW

File: docs/source/developer_guide/add_estimators.rst [hunk 1]
```diff
@@ -133,10 +133,30 @@ or third party modules that aim for interface compliance with ``sktime``,
...
+
+    >>> from sktime.utils.estimator_checks import check_estimator
+    >>> from sktime.forecasting.naive import NaiveForecaster
+    >>> check_estimator(NaiveForecaster)
+
 *   importing test classes, e.g., ``test_all_estimators.TestAllEstimators`` or
     ``test_all_forecasters.TestAllForecasters``. The imports will be discovered directly
     by ``pytest``. The test suite also be extended by inheriting from the test classes.
 
+.. code-block:: python
```
Comment (fkiraly) [inline docs/source/developer_guide/add_estimators.rst:147]: this is not how you would do it if you want to run it via pytest.

pytest runs them if the class is *present* already, it does not need to be run.

**Review by fkiraly — CHANGES_REQUESTED** [review]
These are useful code blocks!

Although, the additions may be in the wrong place.

The section you added them in is using the `sktime` tests as part of a third party test suite. But, the examples are about running the tests locally, so should be perhaps in that section?

In a test suite, either `check_estimator` has to be called within a test. Or, the classes need to be imported, but no methods need to becalled.

File: docs/source/developer_guide/add_estimators.rst (hunk 1 above)
Comment (aiwalter) [reply docs/source/developer_guide/add_estimators.rst:147]: can you give the code snippet for `pytest`?

Comment (fkiraly) [reply docs/source/developer_guide/add_estimators.rst:147]: sth along the lines of

```python

from sktime.tests.test_all_estimators import TestAllEstimators

from mypackage.registry import list_of_all_estimators_getter()

class TestEstimatorsInMyPackage(TestAllEstimators):

    # this overrides the retrieval of estimators to the third party package
    def _all_estimators(self):
        return list_of_all_estimators_getter()

    def test_my_new_custom_test(self, estimator_class, scenario):
        # do sth useful here, fixtures available are as in BaseFixtureGenerator
```

to override fixture generation or add more fixtures, one needs to add `_generate_` functions; decorating a test with a `pytest.mark.parametrize` also works.

Comment (fkiraly) [reply docs/source/developer_guide/add_estimators.rst:147]: this is the complicate route, for extending the *test framework*

Comment (fkiraly) [reply docs/source/developer_guide/add_estimators.rst:147]: the easier route is sth like

```python

from sktime.utils.estimator_checks import check_estimator

@pytest.mark.parametrize("estimator", ESTIMATOR_LIST)
def test_my_estimators(estimator):
    check_estimator(estimator, return_exception=False)
```

Comment (aiwalter) [reply docs/source/developer_guide/add_estimators.rst:147]: I just wanted to document somewhere how to run test for a dedicated estimator, so the last one seems good for that

Comment (fkiraly) [reply docs/source/developer_guide/add_estimators.rst:147]: but, this is for a test in an external package.

If you are *developing*, you just call `check_estimator`

**Comment by fkiraly** [comment]
@aiwalter, I cleaned this up a bit and added some details.

Kindly have a look, I hope you concur that this is an improvement and can go in?

**Pull request merged**